src/app/launcher/launcher.component.ts

```diff
@@ -34,6 +34,11 @@ export class LauncherComponent implements AfterViewInit, OnInit {
    */
   @Input() targetEnvironment: string;
 
+  /**
+   * Get the space name from the space
+   */
+  @Input() spaceName: string;
```

Comment:
Instead of providing an input, I envisioned we would use DemoDependencyCheckService. That way, OSIO and Launcher can have separate implementations -- the launcher component is also cleaner. 

FYI, OSIO uses a Context object to provide the current space. You can find plenty of examples in OSIO, but obtaining a space looks something like this, below. My thinking was that the DemoDependencyCheckService would contain similar code for OSIO.

import { Context } from 'ngx-fabric8-wit';

export class MySpacesComponent {

  private context: Context;

  constructor(private contexts: Contexts) {
    this.subscriptions.push(this.contexts.current.subscribe(val => this.context = val));
  }
...

Reply:
Yes, this could be done here. To my knowledge, this process is already happening in fabric8-ui where the launcher component is called. So i thought of passing the value to the launcher component.

Reply:
It really can be done either way, but prefer to follow the OSIO approach. Typically, the OSIO components subscribe to the Context themselves in order to obtain an observable. The space name shouldn't change, but the observable helps ensure we don't have stale data.

Reply:
okay. i will go with the OSIO approach.